src/Libraries/Microsoft.Extensions.AI.Ollama/OllamaChatClient.cs

```diff
@@ -247,11 +253,36 @@ private static ChatMessage FromOllamaMessage(OllamaChatResponseMessage message)
         return new ChatMessage(new(message.Role), contents);
     }
 
+    private static FunctionCallContent ToFunctionCallContent(OllamaFunctionToolCall function)
+    {
+        var id = Guid.NewGuid().ToString().Substring(0, 8);
+        return new FunctionCallContent(id, function.Name, function.Arguments);
+    }
+
+    private static JsonElement? ToOllamaChatResponseFormat(ChatResponseFormat? format)
+    {
+        if (format is ChatResponseFormatJson jsonFormat)
+        {
+            if (!string.IsNullOrEmpty(jsonFormat.Schema))
+            {
+                return JsonDocument.Parse(jsonFormat.Schema!).RootElement;
```

Comment:
This is going to leak an ArrayPool array. Can you use `JsonSerializer.Deserialize<JsonElement>` instead?

Reply:
Thanks for the tip, but please see my other PR comment before we decide on this.

Reply:
Resolving as no longer applicable.